source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -3264,6 +3270,131 @@ let rec parse_statement
             unexpected_models;
           };
       ]
+  | {
+   Node.value =
+     If
+       {
+         If.body;
+         If.test =
+           {
+             Node.value =
+               ComparisonOperator
+                 {
+                   ComparisonOperator.right = { Node.value = Tuple tuple; _ };
+                   ComparisonOperator.operator;
+                   _;
+                 };
+             _;
+           };
+         If.orelse;
+       };
+   location;
+  } ->
+      let version_list =
+        List.map
+          ~f:(fun expr ->
+            match expr with
+            | { Node.value = Constant const; _ } -> (
+                match const with
+                | Integer integer -> integer
+                | _ -> 0)
+            | _ -> 0)
+          tuple
+      in
+      let major, minor, micro =
+        match version_list with
+        | [first; second; third; _] -> first, second, third
+        | [first; second; third] -> first, second, third
+        | [first; second] -> first, second, 0
+        | [first] -> first, 0, 0
+        | _ -> 0, 0, 0
+      in
+      let extract_version some_version =
+        match some_version with
+        | None -> 0
+        | Some n -> n
+      in
+      let major_version = extract_version python_major_version in
+      let minor_version = extract_version python_minor_version in
+      let micro_version = extract_version python_micro_version in
+      let evaluation_result =
+        match operator with
+        | Equals ->
+            if major_version == major && minor_version == minor && micro_version == micro then
+              true
+            else
+              false
+        | NotEquals ->
+            if major_version != major || minor_version != minor || micro_version != micro then
+              true
+            else
+              false
+        | GreaterThan ->
+            if major_version > major then
+              true
+            else if major_version == major then
+              if minor_version > minor then
+                true
+              else if minor_version == minor then
+                if micro_version > micro then true else false
+              else
+                false
+            else
+              false
+        | GreaterThanOrEquals ->
+            if major_version > major then
+              true
+            else if major_version == major then
+              if minor_version > minor then
+                true
+              else if minor_version == minor then
+                if micro_version >= micro then true else false
+              else
+                false
+            else
+              false
```

Comment:
I wonder if we can compute this by combing the logic for determining `Equals` and `GreaterThan`, since `GreaterThanOrEquals` is equivalent to `Equals` or `GreaterThan`?

Reply:
Yes!